core/src/main/scala/org/apache/spark/storage/DiskBlockObjectWriter.scala

```diff
@@ -174,23 +174,46 @@ private[spark] class DiskBlockObjectWriter(
    * Should call after committing or reverting partial writes.
    */
   private def closeResources(): Unit = {
-    if (initialized) {
-      Utils.tryWithSafeFinally {
-        mcs.manualClose()
-      } {
-        channel = null
-        mcs = null
-        bs = null
-        fos = null
-        ts = null
-        objOut = null
-        initialized = false
-        streamOpen = false
-        hasBeenClosed = true
+    try {
+      if (streamOpen) {
+        Utils.tryWithSafeFinally {
+          objOut = closeIfNonNull(objOut)
+          bs = null
+        } {
+          bs = closeIfNonNull(bs)
```

Comment:
```suggestion
          if (null != objOut) objOut.close()
          bs = null
        } {
          if (null != bs) bs.close()
          bs = null
```

And remove `closeIfNonNull`

Reply:
ok. I'll fix it.